Source/JavaScriptCore/runtime/JSLock.cpp

```diff
@@ -155,6 +155,10 @@ void JSLock::didAcquireLock()
             samplingProfiler->noticeJSLockAcquisition();
     }
 #endif
+#if ASSERT_ENABLED || OS(LINUX)
+    StackBounds::currentThreadStackBounds(); // Check stack bounds consistency.
+#endif
```

Comment:
Yeah, it seems it is too slow. So I think we should just enable it only when `ASSERT_ENABLED`.